src/openrct2-ui/windows/Ride.cpp

```diff
@@ -3311,6 +3369,53 @@ static void WindowRideOperatingUpdate(rct_window* w)
     }
 }
 
+static void WindowRideOperatingTextinput(rct_window* w, rct_widgetindex widgetIndex, char* text)
+{
+    if (text == nullptr)
+        return;
+
+    auto ride = get_ride(w->rideId);
+    if (ride == nullptr)
+        return;
+
+    if (widgetIndex == WIDX_MODE_TWEAK)
+    {
+        const auto& operatingSettings = ride->GetRideTypeDescriptor().OperatingSettings;
+        uint8_t maxValue = gCheatsUnlockOperatingLimits ? OpenRCT2::Limits::CheatsMaxOperatingLimit
+                                                        : operatingSettings.MaxValue;
+        uint8_t minValue = gCheatsUnlockOperatingLimits ? 0 : operatingSettings.MinValue;
+
+        try
+        {
+            uint8_t size = std::stol(std::string(text));
+            size = std::clamp(size, minValue, maxValue);
+            set_operating_setting(ride->id, RideSetSetting::Operation, size);
+        }
+        catch (const std::logic_error&)
+        {
+            // std::stol can throw std::out_of_range or std::invalid_argument
+        }
+    }
+    else if (widgetIndex == WIDX_MINIMUM_LENGTH || widgetIndex == WIDX_MAXIMUM_LENGTH)
+    {
+        try
+        {
+            auto rideSetSetting = widgetIndex == WIDX_MINIMUM_LENGTH ? RideSetSetting::MinWaitingTime
+                                                                     : RideSetSetting::MaxWaitingTime;
+
+            uint16_t upper_bound = OpenRCT2::Limits::MaxWaitingTime;
+            uint16_t lower_bound = 0;
```

Comment:
Snake case should not be used. It is only in old code that hasn't been refactored.

Reply:
Try keep the two branches as similar as possible in terms of layout, variable name, etc.